### PR TITLE
Tags field type

### LINKED ISSUE #1269: Field Type: Tags
Make tags it's own field type to go along side multi_select, key_select, etc
It currently rides off of the multi_select field type.
the multi_select is design to be a finite list of items. 
The tags field is meant to have many options and lets the user add options.

This will let a record have multiple different tag fields. 

Bonus points for also implementing this issue at the same time: #567 
Normal tags are stored in the `postmeta` table.
Private tags would be stored in the `dt_post_user_meta`

### PR DESCRIPTION
**Pull request by cairocoder01**
Resolves #1269 

Adds `tags` field type and converts the existing tags field to use it.

### CODE REVIEW

**Comment by corsacca** [comment]
Great work @cairocoder01 !! :) This all looks good.
I pushed 2 small fixes as i tested it out.

One spot i forgot to mention to look for is: the create new record page. The tag fields now show up there but we have no js to run the search.

**Comment by cairocoder01** [comment]
@corsacca Does your comment mean that you have fixed the new record page, or do I need to check and confirm that since you said there is no js to run the search?

**Comment by corsacca** [comment]
> @corsacca Does your comment mean that you have fixed the new record page, or do I need to check and confirm that since you said there is no js to run the search?

Tags on the new record page still need to be implemented

**Comment by cairocoder01** [comment]
@corsacca This should be ready now

File: dt-core/global-functions.php [hunk 1]
```diff
@@ -422,11 +422,35 @@ function render_field_for_display( $field_key, $fields, $post, $show_extra_contr
                         </option>
                     <?php endforeach; ?>
                 </select>
+            <?php elseif ( $field_type === "tags" ) : ?>
+                <div class="tags">
+                    <var id="<?php echo esc_html( $display_field_id ); ?>-result-container" class="result-container"></var>
+                    <div id="tags_t" name="form-tags" class="scrollable-typeahead typeahead-margin-when-active">
```
Comment (corsacca) [inline dt-core/global-functions.php:428]: I think id needs to be the $display_field_id too, so we don't have multiple fields with the same id

Comment (corsacca) [reply dt-core/global-functions.php]: Ha. That was fast :)

**Pull request merged**